lib/systems/default.nix

```diff
@@ -283,7 +283,7 @@ let
           # to an emulator program. That is, if an emulator requires additional
           # arguments, a wrapper should be used.
           if pkgs.stdenv.hostPlatform.canExecute final
-          then pkgs.writeShellScript "exec" ''exec "$@"''
+          then lib.getExe (pkgs.writeShellScriptBin "exec" ''exec "$@"'')
```

Comment:
I guess more direct would be `"${pkgs.writeShellScript "exec" ''exec "$@"'}"`. Would probably make the intention clearer, but realistically no difference.